llvm/test/Object/archive-malformed-object.test

```diff
@@ -9,37 +10,52 @@
 # RUN: llvm-as input.ll -o input.bc
 # RUN: cp input.bc good.bc
 # RUN: %python -c "with open('input.bc', 'a') as f: f.truncate(10)"
-# RUN: not llvm-ar rc bad.a input.bc 2>&1 | FileCheck %s --check-prefix=ERR1
+# RUN: llvm-ar rc bad.a input.bc 2>&1 | FileCheck %s --check-prefix=WARN1
```

Comment:
llvm-ar.cpp:1045 says the default archive format is decided by the default target triple. If it is COFF/AIXBIG/DARWIN, there would still be an error.

We need to enumerate different archive formats...
```
rm -f bad.a && llvm-ar --format=gnu rc bad.a input.bc
rm -f bad.a && llvm-ar --format=darwin rc bad.a input.bc
...
```

Reply:
Done

Reply:
Maybe I'm missing something, but I don't see this being addressed?

Reply:
Yeah, I had done the second part but not the first. Done now.